discrete RL examples

Could you add a minimal example for discrete action space RL?

I tried to tweak `baselines/vpg/experiment.py` to run `CartPole-v0` but I get the following error:
```
[carlo@x1 baselines]$ python vpg/experiment.py 
Logging directory '/home/carlo/Git/lagom/baselines/logs/default' already existed, do you want to clean it ?
>>> y
@ Experiment: ID: 0 (1), Seed: 1770966829, Device: cpu, Job: 0 (3), PID: 3352
##################################################
# ID: 0
# log.freq: 10
# checkpoint.num: 3
# env.id: CartPole-v0
# env.standardize_obs: True
# env.standardize_reward: True
# nn.sizes: [64, 64]
# agent.lr: 0.001
# agent.use_lr_scheduler: False
# agent.gamma: 0.99
# agent.gae_lambda: 0.97
# agent.standardize_adv: True
# agent.max_grad_norm: 0.5
# agent.entropy_coef: 0.01
# agent.value_coef: 0.5
# env.clip_action: False
# agent.std0: 0.6
# train.timestep: 1000000
# train.timestep_per_iter: 1000
##################################################
@ Experiment: ID: 0 (1), Seed: 1500925526, Device: cpu, Job: 1 (3), PID: 3353
##################################################
# ID: 0
# log.freq: 10
# checkpoint.num: 3
# env.id: CartPole-v0
# env.standardize_obs: True
# env.standardize_reward: True
# nn.sizes: [64, 64]
# agent.lr: 0.001
# agent.use_lr_scheduler: False
# agent.gamma: 0.99
# agent.gae_lambda: 0.97
# agent.standardize_adv: True
# agent.max_grad_norm: 0.5
# agent.entropy_coef: 0.01
# agent.value_coef: 0.5
# env.clip_action: False
# agent.std0: 0.6
# train.timestep: 1000000
# train.timestep_per_iter: 1000
##################################################
@ Experiment: ID: 0 (1), Seed: 2054191100, Device: cpu, Job: 2 (3), PID: 3354
##################################################
# ID: 0
# log.freq: 10
# checkpoint.num: 3
# env.id: CartPole-v0
# env.standardize_obs: True
# env.standardize_reward: True
# nn.sizes: [64, 64]
# agent.lr: 0.001
# agent.use_lr_scheduler: False
# agent.gamma: 0.99
# agent.gae_lambda: 0.97
# agent.standardize_adv: True
# agent.max_grad_norm: 0.5
# agent.entropy_coef: 0.01
# agent.value_coef: 0.5
# env.clip_action: False
# agent.std0: 0.6
# train.timestep: 1000000
# train.timestep_per_iter: 1000
##################################################
concurrent.futures.process._RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/usr/lib/python3.7/concurrent/futures/process.py", line 232, in _process_worker
    r = call_item.fn(*call_item.args, **call_item.kwargs)
  File "/usr/lib/python3.7/concurrent/futures/process.py", line 191, in _process_chunk
    return [fn(*args) for args in chunk]
  File "/usr/lib/python3.7/concurrent/futures/process.py", line 191, in <listcomp>
    return [fn(*args) for args in chunk]
  File "/home/carlo/.local/lib/python3.7/site-packages/lagom-0.0.2-py3.7.egg/lagom/utils/serialize.py", line 96, in __call__
    return self.x(*args, **kwargs)
  File "/home/carlo/.local/lib/python3.7/site-packages/lagom-0.0.2-py3.7.egg/lagom/experiment/run_experiment.py", line 128, in _run
    result = run(config, seed, device, logdir)
  File "vpg/experiment.py", line 89, in run
    train_logger = engine.train(i)
  File "../baselines/vpg/engine.py", line 18, in train
    D = self.runner(self.agent, self.env, self.config['train.timestep_per_iter'])
  File "/home/carlo/.local/lib/python3.7/site-packages/lagom-0.0.2-py3.7.egg/lagom/runner.py", line 54, in __call__
    next_observation, [reward], [step_info] = env.step(action)
  File "/home/carlo/.local/lib/python3.7/site-packages/lagom-0.0.2-py3.7.egg/lagom/envs/wrappers/vec_step_info.py", line 42, in step
    observations, rewards, dones, infos = self.env.step(actions)
  File "/home/carlo/.local/lib/python3.7/site-packages/lagom-0.0.2-py3.7.egg/lagom/envs/wrappers/vec_standardize_reward.py", line 53, in step
    observations, rewards, dones, infos = self.env.step(actions)
  File "/home/carlo/.local/lib/python3.7/site-packages/lagom-0.0.2-py3.7.egg/lagom/envs/wrappers/vec_standardize_observation.py", line 37, in step
    observations, rewards, dones, infos = self.env.step(actions)
  File "/home/carlo/.local/lib/python3.7/site-packages/lagom-0.0.2-py3.7.egg/lagom/envs/wrappers/vec_monitor.py", line 22, in step
    observations, rewards, dones, infos = self.env.step(actions)
  File "/home/carlo/.local/lib/python3.7/site-packages/lagom-0.0.2-py3.7.egg/lagom/envs/vec_env.py", line 65, in step
    observation, reward, done, info = env.step(action)
  File "/home/carlo/.local/lib/python3.7/site-packages/gym/wrappers/time_limit.py", line 15, in step
    observation, reward, done, info = self.env.step(action)
  File "/home/carlo/.local/lib/python3.7/site-packages/gym/envs/classic_control/cartpole.py", line 92, in step
    assert self.action_space.contains(action), "%r (%s) invalid"%(action, type(action))
AssertionError: 1.0 (<class 'numpy.float64'>) invalid
```

--- a/baselines/cem/agent.py
+++ b/baselines/cem/agent.py
@@ -56,7 +56,7 @@ class Agent(BaseAgent):
         action_dist = self.action_head(features)
         out['entropy'] = action_dist.entropy()
         action = action_dist.sample()
-        out['raw_action'] = numpify(action, 'float')
+        out['raw_action'] = numpify(action, self.env.action_space.dtype)
         return out
     
     def learn(self, D, **kwargs):

--- a/baselines/cmaes/agent.py
+++ b/baselines/cmaes/agent.py
@@ -56,7 +56,7 @@ class Agent(BaseAgent):
         action_dist = self.action_head(features)
         out['entropy'] = action_dist.entropy()
         action = action_dist.sample()
-        out['raw_action'] = numpify(action, 'float')
+        out['raw_action'] = numpify(action, self.env.action_space.dtype)
         return out
     
     def learn(self, D, **kwargs):

--- a/baselines/openaies/agent.py
+++ b/baselines/openaies/agent.py
@@ -56,7 +56,7 @@ class Agent(BaseAgent):
         action_dist = self.action_head(features)
         out['entropy'] = action_dist.entropy()
         action = action_dist.sample()
-        out['raw_action'] = numpify(action, 'float')
+        out['raw_action'] = numpify(action, self.env.action_space.dtype)
         return out
     
     def learn(self, D, **kwargs):

--- a/baselines/ppo/agent.py
+++ b/baselines/ppo/agent.py
@@ -102,7 +102,7 @@ class Agent(BaseAgent):
         
         action = action_dist.sample()
         out['action'] = action
-        out['raw_action'] = numpify(action, 'float')
+        out['raw_action'] = numpify(action, self.env.action_space.dtype)
         out['action_logprob'] = action_dist.log_prob(action.detach())
         
         V = self.value(obs)

--- a/baselines/vpg/agent.py
+++ b/baselines/vpg/agent.py
@@ -76,7 +76,7 @@ class Agent(BaseAgent):
         
         action = action_dist.sample()
         out['action'] = action
-        out['raw_action'] = numpify(action, 'float')
+        out['raw_action'] = numpify(action, self.env.action_space.dtype)
         out['action_logprob'] = action_dist.log_prob(action.detach())
         
         V = self.V_head(features)

